switch to new jit_builder.

xuhancn opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #118515
* #118514



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

Differential Revision: [D53203201](https://our.internmc.facebook.com/intern/diff/D53203201)

Skylion007 commented on torch/_inductor/codecache.py:
```diff
@@ -1079,7 +1009,62 @@ def __bool__(self) -> bool:  # type: ignore[override]
+    from filelock import FileLock
+
+    lock_dir = get_lock_dir()
+    lock = FileLock(os.path.join(lock_dir, key + ".lock"), timeout=LOCK_TIMEOUT)
+    with lock:
```
```suggestion
    with FileLock(os.path.join(lock_dir, key + ".lock"), timeout=LOCK_TIMEOUT) as lock:
```

Skylion007 commented on torch/_inductor/codecache.py:
```diff
@@ -1834,31 +1882,37 @@ def _load_library(cls, path: str, key: str) -> Union[CDLL, ModuleType]:
 
             lock_dir = get_lock_dir()
             lock = FileLock(os.path.join(lock_dir, key + ".lock"), timeout=LOCK_TIMEOUT)
             with lock:
```
same here

xuhancn replied:
Done

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D53203201).

xuhancn committed:
switch to new jit_builder.

[ghstack-poisoned]

xuhancn committed:
Update on "switch to new jit_builder."




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

Differential Revision: [D53203201](https://our.internmc.facebook.com/intern/diff/D53203201)

[ghstack-poisoned]
